ledger/light/handle/set_request_test.go

```diff
@@ -0,0 +1,299 @@
+		assert.EqualError(t, err, "something strange from checkjet")
+	})
+
+	t.Run("passed flag is false and checkjet returns ErrNotExecutor", func(t *testing.T) {
+		t.Parallel()
+		f := mockProcedures(t, nil, proc.ErrNotExecutor, nil, nil)
+
+		handler := NewSetRequest(emptyDeps(), msg, false)
+		err := handler.Present(ctx, f)
+		require.NoError(t, err)
+	})
+
+	t.Run("passed flag is true and checkjet returns ErrNotExecutor", func(t *testing.T) {
+		t.Parallel()
+		f := mockProcedures(t, nil, proc.ErrNotExecutor, nil, nil)
+
+		handler := NewSetRequest(emptyDeps(), msg, true)
+		err := handler.Present(ctx, f)
+		require.Error(t, err)
+		assert.Equal(t, proc.ErrNotExecutor, err)
+	})
+}
+
+func TestSetRequest_ErrorFromWaitHot(t *testing.T) {
+	t.Parallel()
+	ctx := flow.TestContextWithPulse(
+		inslogger.TestContext(t),
+		insolar.GenesisPulse.PulseNumber+10,
+	)
+
+	msg := correctMetaMsg(t)
+
+	t.Run("waithot procedure returns err", func(t *testing.T) {
+		t.Parallel()
+		f := mockProcedures(t, nil, nil, errors.New("error from waithot"), nil)
+
+		handler := NewSetRequest(emptyDeps(), msg, false)
+		err := handler.Present(ctx, f)
+		assert.EqualError(t, err, "error from waithot")
+	})
+
+	t.Run("waithot procedure returns nil err", func(t *testing.T) {
+		t.Parallel()
+		f := mockProcedures(t, nil, nil, nil, nil)
+
+		handler := NewSetRequest(emptyDeps(), msg, false)
+		err := handler.Present(ctx, f)
+		require.NoError(t, err)
+	})
+}
+
+func TestSetRequest_ErrorFromSetRequest(t *testing.T) {
+	t.Parallel()
+	ctx := flow.TestContextWithPulse(
+		inslogger.TestContext(t),
+		insolar.GenesisPulse.PulseNumber+10,
+	)
+
+	msg := correctMetaMsg(t)
+
+	t.Run("setrequest procedure returns err", func(t *testing.T) {
+		t.Parallel()
+		f := mockProcedures(t, nil, nil, nil, errors.New("error from setrequest"))
+
+		handler := NewSetRequest(emptyDeps(), msg, false)
+		err := handler.Present(ctx, f)
+		assert.EqualError(t, err, "error from setrequest")
+	})
+
+	t.Run("setrequest procedure returns nil err", func(t *testing.T) {
+		t.Parallel()
+		f := mockProcedures(t, nil, nil, nil, nil)
+
+		handler := NewSetRequest(emptyDeps(), msg, false)
+		err := handler.Present(ctx, f)
+		require.NoError(t, err)
+	})
+}
+
+func correctMetaMsg(t *testing.T) payload.Meta {
+	ref := gen.Reference()
+
+	virtual := record.Virtual{
+		Union: &record.Virtual_Request{
+			Request: &record.Request{
+				Object: &ref,
+			},
+		},
+	}
+	virtualBuf, err := virtual.Marshal()
+	require.NoError(t, err)
+
+	request := payload.SetRequest{
+		Request: virtualBuf,
+	}
+	requestBuf, err := request.Marshal()
+	require.NoError(t, err)
+
+	msg := payload.Meta{
+		Payload: requestBuf,
+	}
+
+	return msg
+}
+
+func emptyDeps() *proc.Dependencies {
+	return &proc.Dependencies{
+		CalculateID: func(p *proc.CalculateID) {},
+		CheckJet:    func(p *proc.CheckJet) {},
+		WaitHotWM:   func(p *proc.WaitHotWM) {},
+		SetRequest:  func(p *proc.SetRequest) {},
+	}
+}
+
+func mockProcedures(
```

Comment:
This fucntion has a generic name, but  only contains SetRequest procedures. It will be a problem when more tests are written. Also, it doesnt allow to mock return values (only error). It's probably easier just to mock it for each test separately.

Reply:
Done